__tests__/empty-line-around-horizontal-rules.test.ts

```diff
@@ -0,0 +1,78 @@
+import dedent from 'ts-dedent';
+import EmptyLineAroundHorizontalRules from '../src/rules/empty-line-around-horizontal-rules';
+import {ruleTest} from './common';
+
+ruleTest({
+  RuleBuilderClass: EmptyLineAroundHorizontalRules,
+  testCases: [
+    {
+      testName:
+        'Make sure that basic dash horizontal rules have lines added around them',
+      before: dedent`
+        - asdf
+        ---
+        - qwer
+      `,
+      after: dedent`
+        - asdf
+        ${''}
+        ---
+        ${''}
+        - qwer
+      `,
+    },
+    {
+      testName: 'Make sure that basic asterisk horizontal rules have lines added around them',
+      before: dedent`
+        asdf
+        ***
+        qwer
+      `,
+      after: dedent`
+        asdf
+        ${''}
+        ***
+        ${''}
+        qwer
+      `,
+    },
+    {
+      testName: 'Make sure that basic underscore horizontal rules have lines added around them',
+      before: dedent`
+        asdf
+        ___
+        qwer
+      `,
+      after: dedent`
+        asdf
+        ${''}
+        ___
+        ${''}
+        qwer
+      `,
+    },
+    {
+      testName: 'Make sure that horizontal rules for frontmatter don\'t have lines added around them',
+      before: dedent`
+        ---
+        prop: value
+        ---
+        ${''}
+        asdf
+        ___
+        qwer
+      `,
+      after: dedent`
+        ---
+        prop: value
+        ---
+        ${''}
+        asdf
+        ${''}
+        ___
+        ${''}
+        qwer
+      `,
+    },
```

Comment:
We will want to add some corresponding tests that test how the hrs work in blockquotes and nested blockquotes to make sure that the blank line generation works as intended for those scenarios as well.

Reply:
Is what I added what you're looking for?

Reply:
This is definitely close. With the blockquote scenario, I was more wondering about the following scenarios:
Assuming horizontal rules can be in a blockquote:
``` markdown
> Blockquote
>
> ---
> > Nested Blockquote
> Blockquote
> > Nested Blockquote
```
And the following edge cases (how does it handle blockquotes of the same level with an empty blockquote line between them):
``` markdown
> > Blockquote
> > 
> > ---
>
> > Nested Blockquote
> Blockquote
> > Nested Blockquote

Callout scenario:
> > Blockquote
> > 
> > ---
>
> > [!Info] Nested Blockquote
> > Info here
> Blockquote
> > Nested Blockquote

```